src/components/screens/consul/Proposals/NewProposal.js

```diff
@@ -325,33 +321,78 @@ export const NewProposal = ({ navigation, data, query }) => {
   );
 };
 
-const ImageSelector = ({ control, imageUri, item, onChange, selectImage, value }) => {
-  const { buttonTitle, infoText, label, name } = item;
+const ImageSelector = ({ control, field, item, selectImage }) => {
+  const [errorMessage, setErrorMessage] = useState();
 
-  useEffect(() => {
-    // the `!value` control is used for editing the proposal
-    // the proposal contains an image, it is for showing the preview to the users
-    if (!value) {
-      onChange(imageUri);
-    }
-  }, [imageUri]);
+  const { buttonTitle, infoText } = item;
+  const { name, onChange, value } = field;
 
   return (
     <>
-      <Input hidden label={label} control={control} name={name} value={value} />
+      <Input
+        {...item}
+        control={control}
+        errorMessage={errorMessage}
+        hidden
+        name={name}
+        validate={true}
+        value={value}
+      />
       <RegularText smallest placeholder>
         {infoText}
       </RegularText>
 
       {value ? (
         <WrapperRow center spaceBetween>
           <Image source={{ uri: value }} style={styles.image} />
-          <Touchable onPress={() => onChange(null)}>
+
+          <TouchableOpacity
+            onPress={() => {
+              onChange('');
+              setErrorMessage('');
+            }}
+          >
             <Icon.Trash color={colors.error} size={normalize(16)} />
-          </Touchable>
+          </TouchableOpacity>
         </WrapperRow>
       ) : (
-        <Button title={buttonTitle} invert onPress={selectImage} />
+        <Button
+          title={buttonTitle}
+          invert
+          onPress={async () => {
+            const { uri } = await selectImage();
+            const { size } = await FileSystem.getInfoAsync(uri);
+
+            // `1048576` = the byte equivalent of 1MB
+            if (size > 1048576) {
+              setErrorMessage(
+                texts.consul.startNew['choose-image-must-be-in-between-0-bytes-and-1-mb']
+              );
+            }
+
+            // if the photo that wants to be uploaded is not a jpg,
+            // we show this error message on the screen
+            if (!uri.includes('.jpg')) {
+              setErrorMessage(
+                texts.consul.startNew[
+                  'choose-image-content-type-image/png-does-not-match-any-of-accepted-content-types-jpg'
+                ]
+              );
+            }
+
+            // if both requests are not met, a message with all
+            // errors is displayed on the screen
+            if (!uri.includes('.jpg') && size > 1048576) {
```

Comment:
the checks for `.jpg` and `1048576` are duplicated here. this could be smarter. any ideas?

Reply:
4f4769453427750e74db1b209268861c890b1939